cpu/esp8266/periph/rtt.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2020 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp8266
+ * @ingroup     drivers_periph_rtt
+ * @{
+ *
+ * @file
+ * @brief       Low-level RTT driver implementation for ESP8266
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#include "cpu.h"
+#include "log.h"
+#include "periph/rtt.h"
+
+#include "esp/common_macros.h"
+#include "esp/dport_regs.h"
+#include "esp/rtc_regs.h"
+#include "sdk/sdk.h"
+
+#define FRC2_CLK_DIV_256    2   /* divider for the 80 MHz AHB clock */
+
+#define RTC_BSS_ATTR __attribute__((section(".rtc.bss")))
+
+/**
+ * FRC2 is a 32-bit countup timer, triggers interrupt when reaches alarm value.
+ */
+typedef struct {
+    uint32_t load;
+    uint32_t count;
+    union {
+        struct {
+            uint32_t intr_hold : 1;
+            uint32_t reserved1 : 1;
+            uint32_t clk_div   : 2;
+            uint32_t reserved2 : 2;
+            uint32_t reload    : 1;
+            uint32_t enable    : 1;
+            uint32_t intr_sta  : 1;
+            uint32_t reserved3 : 23;
+        };
+        uint32_t val;
+    } ctrl;
+    union {
+        struct {
+            uint32_t clear    : 1;
+            uint32_t reserved1: 31;
+        };
+        uint32_t val;
+    } intr;
+    uint32_t alarm;
+} frc2_struct_t;
+
+extern volatile frc2_struct_t frc2;
```

Comment:
Ah now I get that linker magic.
I think the more orthodox way would be to write
```C
#define FRC2 ((frc2_struct_t *) 0x60000620)
```
and have the members be `volatile`. Then you wouldn't have to deal with the linker at all.

But if that's how you're handling things on esp*, carry on.

Reply:
> But if that's how you're handling things on `esp*`.

Yeah, but i wasn't my idea. All other peripherals are used by the SDK in that way. Therefore, I extended the linker script for this peripheral in same way, just to be consistent.

Reply:
Can you add a comment? Something like `/* linker script will make sure this points to the hardware register address */` so it appears less magic for the unsuspecting reader.